plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/tasks/PostgreBackupAllSettings.java

```diff
@@ -0,0 +1,262 @@
+
+    public void setEncoding(String encoding) {
+        this.encoding = encoding;
+    }
+
+    public boolean isExportOnlyMetadata() {
+        return exportOnlyMetadata;
+    }
+
+    public void setExportOnlyMetadata(boolean exportOnlyMetadata) {
+        this.exportOnlyMetadata = exportOnlyMetadata;
+    }
+
+    public boolean isExportOnlyGlobals() {
+        return exportOnlyGlobals;
+    }
+
+    public void setExportOnlyGlobals(boolean exportOnlyGlobals) {
+        this.exportOnlyGlobals = exportOnlyGlobals;
+    }
+
+    public boolean isExportOnlyRoles() {
+        return exportOnlyRoles;
+    }
+
+    public void setExportOnlyRoles(boolean exportOnlyRoles) {
+        this.exportOnlyRoles = exportOnlyRoles;
+    }
+
+    public boolean isExportOnlyTablespaces() {
+        return exportOnlyTablespaces;
+    }
+
+    public void setExportOnlyTablespaces(boolean exportOnlyTablespaces) {
+        this.exportOnlyTablespaces = exportOnlyTablespaces;
+    }
+
+    public boolean isNoPrivileges() {
+        return noPrivileges;
+    }
+
+    public void setNoPrivileges(boolean noPrivileges) {
+        this.noPrivileges = noPrivileges;
+    }
+
+    public boolean isNoOwner() {
+        return noOwner;
+    }
+
+    public void setNoOwner(boolean noOwner) {
+        this.noOwner = noOwner;
+    }
+
+    public boolean isAddRolesPasswords() {
+        return addRolesPasswords;
+    }
+
+    public void setAddRolesPasswords(boolean addRolesPasswords) {
+        this.addRolesPasswords = addRolesPasswords;
+    }
+
+    @Override
+    public void fillExportObjectsFromInput() {
+        PostgreDataSource dataSource = null;
+        List<PostgreDatabase> databases = new ArrayList<>();
+        for (DBSObject object : getDatabaseObjects()) {
+            if (object instanceof PostgreDatabase) {
+                PostgreDatabase database = (PostgreDatabase) object;
+                dataSource = database.getDataSource();
+                databases.add(database);
+            }
+        }
+        if (dataSource != null) {
+            exportObjects.add(new PostgreDatabaseBackupAllInfo(dataSource, databases));
+            updateDataSourceContainer();
+        }
+    }
+
+    @Override
+    public void loadSettings(DBRRunnableContext runnableContext, DBPPreferenceStore store) throws DBException {
+        super.loadSettings(runnableContext, store);
+        encoding = store.getString("pg.export.all.encoding");
+        exportOnlyMetadata = store.getBoolean("pg.export.all.exportOnlyMetadata");
+        exportOnlyGlobals = store.getBoolean("pg.export.all.exportOnlyGlobals");
+        exportOnlyRoles = store.getBoolean("pg.export.all.exportOnlyRoles");
+        exportOnlyTablespaces = store.getBoolean("pg.export.all.exportOnlyTablespaces");
+        noPrivileges = store.getBoolean("pg.export.all.noPrivileges");
+        noOwner = store.getBoolean("pg.export.all.noOwner");
+        addRolesPasswords = store.getBoolean("pg.export.all.addRolesPasswords");
```

Comment:
Can these string values be a constants? In PostgreConstants, for example.

Reply:
Just constants are ok, I think. It is used here only.